### PR TITLE
[Driver] Provide executed command as an actual array

### PR DESCRIPTION
**Pull request by jakepetroules**
This patch adds additional entries to the JSON command messages output
by the Swift compiler. It's now possible to get the command executable
("command_executable") and arguments ("command_arguments") as a single
string and array, respectively, rather than having to parse the
shell-escaped command line provided in the "command" key.

<rdar://problem/35701809>

### CODE REVIEW

File: lib/Driver/ParseableOutput.cpp [hunk 1]
```diff
@@ -98,12 +98,18 @@ class CommandBasedMessage : public Message {
 };
 
 class DetailedCommandBasedMessage : public CommandBasedMessage {
+  std::string Executable;
+  SmallVector<StringRef, 16> Arguments;
```
Comment (jrose-apple) [inline lib/Driver/ParseableOutput.cpp]: This seem suspicious. Why have the executable use an owning string, but the arguments use unowned strings?

(The simplest answer here would be to use `std::string` in all cases.)

Comment (jakepetroules) [reply lib/Driver/ParseableOutput.cpp]: Done.

**Review by graydon — COMMENTED** [review]
I'm not 100% sure which programs (if any) are sensitive to this change; have you tested it with known consumers of parseable-output?

**Comment by jakepetroules** [comment]
@graydon It's a JSON dictionary, so I cannot imagine adding new keys would cause any problems for existing clients. It's purposefully intended to be backwards-compatible for this reason.

**Comment by graydon** [comment]
@jakepetroules oh sorry! I misread and saw the existing key being deleted in the docs and failed to read the code where you're keeping it around.

**Review by graydon — APPROVED** [review]
LGTM

**Comment by jakepetroules** [comment]
@swift-ci Pleas test

**Comment by jakepetroules** [comment]
@swift-ci Please test

**Comment by jakepetroules** [comment]
Fixed the tests!

**Comment by jakepetroules** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
That seems pretty unlikely to be your change.

@swift-ci Please test macOS

**Comment by jakepetroules** [comment]
@jrose-apple Can I merge now? And am I required to choose a particular merge strategy or can I pick either merge or cherry-pick at my discretion?

**Comment by jrose-apple** [comment]
Either Merge or Squash is okay, Rebase is not (because the intermediate commits might not build once rebased, if there's more than one.) Personally I only use Squash if there's just one commit to begin with.

**Comment by jakepetroules** [comment]
> [...] because the intermediate commits might not build once rebased [...]

Interesting point, I never thought about that case before. Previously I was working with Gerrit using a single-commit cherry-pick model (so one "PR" always = one commit) and therefore it wouldn't be possible to run into that issue.

Thanks for helping to land my first change to Swift!

**Pull request merged**